octobot_trading/personal_data/orders/order.py

```diff
@@ -398,13 +419,108 @@ def get_position_side(self, future_contract):
             return enums.PositionSide.BOTH
         raise errors.InvalidPositionSide(f"Can't determine order position side while using hedge position mode")
 
+    @contextlib.contextmanager
+    def active_or_inactive_transition(self):
+        previous_value = self.is_in_active_inactive_transition
+        try:
+            self.is_in_active_inactive_transition = True
+            yield
+        finally:
+            self.is_in_active_inactive_transition = previous_value
+
+    def use_active_trigger(self, active_trigger: price_trigger.PriceTrigger):
+        if active_trigger is None:
+            raise ValueError("active_trigger must be provided")
+        if self.active_trigger is None:
+            self.active_trigger = active_trigger
+        elif self.active_trigger.is_pending():
+            logging.get_logger(self.get_logger_name()).error(
+                f"The current active trigger (at {self.active_trigger.trigger_price}) is still pending, canceling it "
+                f"and replacing it by this new one, this is works but is unexpected."
+            )
+            self.active_trigger.clear()
+            self.active_trigger = active_trigger
+        else:
+            self.active_trigger.trigger_price = active_trigger.trigger_price
+            self.active_trigger.trigger_above = active_trigger.trigger_above
+
+    async def set_as_inactive(self, active_trigger: price_trigger.PriceTrigger):
```

Comment:
```suggestion
    async def set_as_inactive(self, active_trigger: base_trigger.BaseTrigger):
```

Reply:
indeed, it's up, I also added unit tests.